src/nccl_ofi_rdma.c

```diff
@@ -5544,6 +5635,101 @@ static int release_ep(nccl_net_ofi_ep_t *base_ep)
 	return ret;
 }
 
+static nccl_net_ofi_rdma_ep_t *allocate_ep(nccl_net_ofi_rdma_device_t *device)
+{
+	int num_rails = device->num_rails;
+
+	/* Allocate endpoint */
+	nccl_net_ofi_rdma_ep_t *ep = calloc(1, sizeof(nccl_net_ofi_rdma_ep_t));
+	if (!ep) {
+		NCCL_OFI_WARN("Unable to allocate rdma endpoint");
+		return NULL;
+	}
+
+	/* Initialize base endpoint */
+	ep->base.device = &device->base;
+	ep->base.listen = listen;
+	ep->base.connect = connect;
+	ep->base.release_ep = release_ep;
+
+	/* Initialize number of rail */
+	ep->num_rails = num_rails;
+
+	/* Initialize reference count */
+	ep->ref_cnt = 0;
+
+	ep->bounce_buff_size = NCCL_OFI_MAX(NCCL_OFI_MAX(sizeof(nccl_net_ofi_rdma_ctrl_msg_t), eager_max_size),
+						sizeof(nccl_ofi_rdma_connection_info_t));
+
+	NCCL_OFI_TRACE(NCCL_NET, "RDMA endpoint %p for dev #%d is created",
+			ep,
+			device->base.dev_id);
+
+	return ep;
+}
+
+static int initialize_ep(nccl_net_ofi_rdma_ep_t *ep,
```

Comment:
This does more than initialize an endpoint, since it's allocating memory for rails.  I guess i'm not sure why we'd do that here instead of part of alloc?

Reply:
Agree, the distinction was confusing. Consolidated this into one function called `create_ep`.

Long term, I'll create a backlog item to refactor the ownership model of the endpoint. Then we can remove some of the ugliness around `ref_cnt`.